### PR TITLE
[StorageMover] Rename SmbFileShare endpoint cmdlets 

### PR DESCRIPTION
**Pull request by yifanz7**
## Description
Per feature team's ask, 

- rename `New/Update-AzStorageMoverSmbFileShareEndpoint` to `New/Update-AzStorageMoverAzSmbFileShareEndpoint` to avoid confusion
- add the old names as alias. Might remove the alias in a future release.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StorageMover/StorageMover.Autorest/custom/UpdateAzStorageMoverAzSmbFileShareEndpoint.ps1 [hunk 1]
```diff
@@ -53,9 +53,10 @@ INPUTOBJECT<IStorageMoverIdentity>: Identity Parameter
 .Link
 https://learn.microsoft.com/powershell/module/az.storagemover/update-azstoragemoversmbfileshareendpoint
 #>
-function Update-AzStorageMoverSmbFileShareEndpoint {
+function Update-AzStorageMoverAzSmbFileShareEndpoint {
     [OutputType([Microsoft.Azure.PowerShell.Cmdlets.StorageMover.Models.Api20231001.IEndpoint])]
     [CmdletBinding(DefaultParameterSetName = 'UpdateExpanded', PositionalBinding =$false, SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    [Alias("Update-AzStorageMoverSmbFileShareEndpoint")]
```
Comment (blueww) [inline src/StorageMover/StorageMover.Autorest/custom/UpdateAzStorageMoverAzSmbFileShareEndpoint.ps1:59]: We might should add a test case to make sure the old name still work.

File: src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1 [hunk 2]
```diff
@@ -5,7 +5,7 @@ if(($null -eq $TestName) -or ($TestName -contains 'New-AzStorageMoverSmbFileShar
       $loadEnvPath = Join-Path $PSScriptRoot '..\loadEnv.ps1'
   }
   . ($loadEnvPath)
-  $TestRecordingFile = Join-Path $PSScriptRoot 'New-AzStorageMoverSmbFileShareEndpoint.Recording.json'
+  $TestRecordingFile = Join-Path $PSScriptRoot 'New-AzStorageMoverAzSmbFileShareEndpoint.Recording.json'
```
Comment (blueww) [inline src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1:8]: The test case name not changed in line #1

File: src/StorageMover/StorageMover.Autorest/custom/NewAzStorageMoverAzSmbFileShareEndpoint.ps1 [hunk 3]
```diff
@@ -36,9 +36,10 @@ To create the parameters described below, construct a hash table containing the
 .Link
 https://learn.microsoft.com/powershell/module/az.storagemover/new-azstoragemoversmbfileshareendpoint
```
Comment (blueww) [inline src/StorageMover/StorageMover.Autorest/custom/NewAzStorageMoverAzSmbFileShareEndpoint.ps1:37]: help link also need change.

File: src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1 [hunk 4]
```diff
@@ -17,7 +17,7 @@ if(($null -eq $TestName) -or ($TestName -contains 'New-AzStorageMoverSmbFileShar
 Describe 'New/Update-AzStorageMoverSmbFileShareEndpoint' {
```
Comment (blueww) [inline src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1:17]: description need update

File: src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1 (hunk 2 above)
Comment (yifanz7) [reply src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1:8]: Updated

File: src/StorageMover/StorageMover.Autorest/custom/NewAzStorageMoverAzSmbFileShareEndpoint.ps1 (hunk 3 above)
Comment (yifanz7) [reply src/StorageMover/StorageMover.Autorest/custom/NewAzStorageMoverAzSmbFileShareEndpoint.ps1]: Updated

File: src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1 (hunk 4 above)
Comment (yifanz7) [reply src/StorageMover/StorageMover.Autorest/test/New-AzStorageMoverAzSmbFileShareEndpoint.Tests.ps1]: Updated

File: src/StorageMover/StorageMover.Autorest/custom/UpdateAzStorageMoverAzSmbFileShareEndpoint.ps1 (hunk 1 above)
Comment (yifanz7) [reply src/StorageMover/StorageMover.Autorest/custom/UpdateAzStorageMoverAzSmbFileShareEndpoint.ps1:59]: test case added

**Review by blueww — APPROVED** [review]

**Review by dolauli — APPROVED** [review]

**Pull request merged**